### PR TITLE
[PATCH v2] Fix PPC64le cache line issues

### PR DESCRIPTION
**Pull request by lumag**

### CODE REVIEW

File: platform/linux-generic/include/odp_packet_io_internal.h [hunk 1]
```diff
@@ -53,6 +53,8 @@ struct pktio_if_ops;
 
 #if defined(ODP_NETMAP)
 #define PKTIO_PRIVATE_SIZE 74752
+#elif defined(ODP_PKTIO_DPDK) && ODP_CACHE_LINE_SIZE == 128
+#define PKTIO_PRIVATE_SIZE 10240
 #elif defined(ODP_PKTIO_DPDK)
 #define PKTIO_PRIVATE_SIZE 5632
```
[the remark this answers is not in the document]
Comment (lumag) [reply platform/linux-generic/include/odp_packet_io_internal.h]: It's a `sizeof(struct)` aligned up, so it was hand-calculated.

[the remark this answers is not in the document]
Comment (Bill-Fischofer-Linaro) [reply platform/linux-generic/include/odp_packet_io_internal.h]: Can we then use that `sizeof()` in the `#define`?

[the remark this answers is not in the document]
Comment (muvarov) [reply platform/linux-generic/include/odp_packet_io_internal.h]: maybe make unsigned char ODP_ALIGNED_CACHE pkt_priv[PKTIO_PRIVATE_SIZE]; dynamic rather then fixed len? Remove static assert from pkios and increase private date with realloc() is needed? All this ifdefs are very annoying.

[the remark this answers is not in the document]
Comment (lumag) [reply platform/linux-generic/include/odp_packet_io_internal.h]: @Bill-Fischofer-Linaro No, structs are private to drivers.
@muvarov I'd like to hear about realloc for shm-allocated memory.
Another approach might be to allocate pktio-private memory from per-pktio-type pool. However I can not predict impact on the performance.
@matiaselo @psavol any ideas?

[the remark this answers is not in the document]
Comment (lumag) [reply platform/linux-generic/include/odp_packet_io_internal.h]: @muvarov this memory will not be visible to other ODP threads in process mode.

[the remark this answers is not in the document]
Comment (muvarov) [reply platform/linux-generic/include/odp_packet_io_internal.h]: @lumag memory might be not shared for priv and it should work. 1) if pktio open in control thread then workers run - it's the same memory 2) with workers run and do pktio_open then they have per worker internal non share priv struct which will be cleared after close.

[the remark this answers is not in the document]
Comment (muvarov) [reply platform/linux-generic/include/odp_packet_io_internal.h]: API does not forbid to call open several times for the same pool. odp_pktio_open("eth0", pool, NULL) and then odp_pktio_open("netmap:eth0", pool, NULL) or odp_pktio_open("tap0", pool, NULL). Not sure that I understand what "the per-pktio private data pool" means.

**Comment by lumag** [comment]
I meant a pool that will return private data areas for each PtkIO type.
It is not possible to use plain dynamic memory since an application can
access single pktio from different threads, which may have different
memory areas.

**Comment by muvarov** [comment]
@lumag yes, agree. pktio can have internal things which has to be common for all threads. Now I think that it's better to apply current PR as it fixes real issue. And we can polish memory usage later.

File: platform/linux-generic/include/odp_packet_io_internal.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (muvarov) [reply platform/linux-generic/include/odp_packet_io_internal.h]: @matiaselo  only memory usage 10k instead of 5k multiply on ODP_CONFIG_PKTIO_ENTRIES

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**